File: API/src/main/java/org/example/data/dto/AnalyseResponse.java
package org.example.data.dto;

import lombok.*;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AnalyseResponse {
    private List<Double> probabilities;
    private Double mean;
    private Double stdDev;
    private Double coefVariation;
    private Double coefSharp;
    private Double coefInformation;
    private Double coefSortino;
}

File: Analyse/src/main/kotlin/com/example/dto/AnalyseResponse.kt
package com.example.dto

import com.example.utilities.InstantSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.Instant

@Serializable
data class AnalyseResponse(
    @SerialName("profitability")
    val profitability: List<Double>,

    @SerialName("mean")
    val mean: Double,

    @SerialName("std_dev")
    val stdDev: Double,

    @SerialName("coef_variation")
    val coefVariation: Double,

    @SerialName("coef_sharp")
    val coefSharp: Double,

    @SerialName("coef_information")
    val coefInformation: Double,

    @SerialName("coef_sortino")
    val coefSortino: Double,
)
